src/java.base/share/classes/sun/launcher/LauncherHelper.java

```diff
@@ -312,12 +318,104 @@ private static void printLocales() {
+    }
+
+    private static void printSecuritySettings(String arg) {
+        if (arg.toLowerCase(Locale.ROOT).equals("properties")) {
+            printSecurityProperties();
+        } else if(arg.toLowerCase(Locale.ROOT).equals("providers")) {
+            printSecurityProviderConfig();
+        } else if(arg.toLowerCase(Locale.ROOT).equals("tls")) {
+            printSecurityTLSConfig();
+        } else {
+            printSecurityProperties();
+            printSecurityProviderConfig();
+            printSecurityTLSConfig();
+        }
+    }
+
+    private static void printSecurityProperties() {
+        ostream.println(INDENT + "Security properties:");
+        Properties p = SharedSecrets.getJavaSecurityPropertiesAccess().getInitialProperties();
+        for (String key : p.stringPropertyNames().stream().sorted().toList()) {
+            String val = p.getProperty(key);
+            if (val.contains(",") && val.length() > 60) {
+                // split lines longer than 60 chars which have multiple values
+                ostream.println(TWOINDENT + key + "=");
+                List.of(val.split(",")).forEach(s -> ostream.println(THREEINDENT + s.trim()));
+            } else {
+                ostream.println(TWOINDENT + key + "=" + val);
+            }
+        }
+        ostream.print("\n");
+    }
+
+    private static void printSecurityTLSConfig() {
+        SSLSocket ssls;
+        try {
+            ssls = (SSLSocket)
+                    SSLContext.getDefault().getSocketFactory().createSocket();
+        } catch (IOException | NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+
+        ostream.println(INDENT + "Security TLS configuration:");
+        ostream.println(TWOINDENT + "Enabled Protocols:");
+        for (String s : ssls.getEnabledProtocols()) {
+            System.out.println(THREEINDENT + s);
+        }
+
+        System.out.println("\n" + TWOINDENT + "Enabled Cipher Suites:");
+        for (String s : ssls.getEnabledCipherSuites()) {
+            System.out.println(THREEINDENT + s);
+        }
+        ostream.print("\n");
+    }
+
+    private static void printSecurityProviderConfig() {
+        ostream.println(INDENT + "Security provider static configuration:");
+        for (Provider p : Security.getProviders()) {
+            ostream.println(TWOINDENT + "Provider name: " + p.getName());
+            ostream.println(TWOINDENT + "Provider information: " + wrappedString(p.getInfo(), 80));
+            ostream.println(TWOINDENT + "Provider services: (type : algorithm)");
+            for (Provider.Service ps : p.getServices().stream()
+                    .sorted(Comparator.comparing(Provider.Service::getType)).toList()) {
+                ostream.println(THREEINDENT + ps.getType() + " : " + ps.getAlgorithm());
+            }
+        }
+        ostream.print("\n");
+    }
+
+    // return a string split across multiple lines which aims to limit max width
```

Comment:
This utility method might be usable in more places if the indentation was an argument instead of hard coded.
For example, in `printSecurityProperties`.  It might need to be able to split in more places.